src/org/rust/lang/core/psi/util/RustModExtensions.kt

```diff
@@ -0,0 +1,61 @@
+package org.rust.lang.core.psi.util
+
+import com.intellij.psi.PsiDirectory
+import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RustCompositeElement
+import org.rust.lang.core.psi.RustCrateItem
+import org.rust.lang.core.psi.RustModItem
+
+
+private val RustModItem.modDir: PsiDirectory?
+    get() {
+        val parent = parentModInFile ?: return containingFile.parent
+
+        // FIXME(matklad): mod name should be not null, should we rearange our rules for file/mod?
```

Comment:
rules in `rust.bnf`. 

At the moment, out main rule for rust file is 

```
crate_item ::= [ SHEBANG_LINE ] [ inner_attrs ] item * {
  extends = mod_item
}
```

It does not include `IDENTIFIER`, so `getIdentifier` and `getName` for `RustModItem` are nullable. 

But every `mod` in Rust does have a name. 

Hm, actually the root module is nameless... Ok, than it is correct that mod's name it null.